cmd/ncproxy/ncproxy.go

```diff
@@ -0,0 +1,444 @@
+		ID:        ep.Id,
+		Name:      ep.Name,
+		Network:   ep.HostComputeNetwork,
+		Namespace: ep.HostComputeNamespace,
+	}, nil
+}
+
+func (s *grpcService) GetEndpoints(ctx context.Context, req *ncproxygrpc.GetEndpointsRequest) (*ncproxygrpc.GetEndpointsResponse, error) {
+	log.G(ctx).Info("GetEndpoints request")
+
+	rawEndpoints, err := hcn.ListEndpoints()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get HNS endpoints: %s", err)
+	}
+
+	endpoints := make([]*ncproxygrpc.GetEndpointResponse, len(rawEndpoints))
+	for i, endpoint := range rawEndpoints {
+		resp := &ncproxygrpc.GetEndpointResponse{
+			ID:        endpoint.Id,
+			Name:      endpoint.Name,
+			Network:   endpoint.HostComputeNetwork,
+			Namespace: endpoint.HostComputeNamespace,
+		}
+		endpoints[i] = resp
+	}
+	return &ncproxygrpc.GetEndpointsResponse{
+		Endpoints: endpoints,
+	}, nil
+}
+
+func (s *grpcService) GetNetwork(ctx context.Context, req *ncproxygrpc.GetNetworkRequest) (*ncproxygrpc.GetNetworkResponse, error) {
+	log.G(ctx).WithFields(logrus.Fields{
+		"networkName": req.Name,
+	}).Info("GetNetwork request")
+
+	if req.Name == "" {
+		return nil, status.Error(codes.InvalidArgument, "received empty field in request")
+	}
+
+	network, err := hcn.GetNetworkByName(req.Name)
+	if err != nil {
+		if _, ok := err.(hcn.NetworkNotFoundError); ok {
+			return nil, status.Errorf(codes.NotFound, "no network with name `%s` found", req.Name)
+		}
+		return nil, fmt.Errorf("failed to get network with name `%s`: %s", req.Name, err)
+	}
+
+	return &ncproxygrpc.GetNetworkResponse{
+		ID:   network.Id,
+		Name: network.Name,
+	}, nil
+}
+
+func (s *grpcService) GetNetworks(ctx context.Context, req *ncproxygrpc.GetNetworksRequest) (*ncproxygrpc.GetNetworksResponse, error) {
+	log.G(ctx).Info("GetNetworks request")
+
+	rawNetworks, err := hcn.ListNetworks()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get HNS networks: %s", err)
+	}
+
+	networks := make([]*ncproxygrpc.GetNetworkResponse, len(rawNetworks))
+	for i, network := range rawNetworks {
+		resp := &ncproxygrpc.GetNetworkResponse{
+			ID:   network.Id,
+			Name: network.Name,
+		}
+		networks[i] = resp
+	}
+
+	return &ncproxygrpc.GetNetworksResponse{
+		Networks: networks,
+	}, nil
+}
+
+// TTRPC service exposed for use by the shim. Holds a mutex for updating map of
+// client connections.
+type ttrpcService struct {
+	m sync.Mutex
+}
+
+func (s *ttrpcService) RegisterComputeAgent(ctx context.Context, req *ncproxyttrpc.RegisterComputeAgentRequest) (*ncproxyttrpc.RegisterComputeAgentResponse, error) {
+	log.G(ctx).WithFields(logrus.Fields{
+		"containerID":  req.ContainerID,
+		"agentAddress": req.AgentAddress,
+	}).Info("RegisterComputeAgent request")
+
+	conn, err := winio.DialPipe(req.AgentAddress, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect to compute agent service")
+	}
+	cl := ttrpc.NewClient(conn, ttrpc.WithOnClose(func() { conn.Close() }))
```

Comment:
nit: why is this variable named like this?

Reply:
No real reason, just short for client. I'll change to `client` and in the other two places this is used also